src/transformers/training_args.py

```diff
@@ -1794,6 +1794,13 @@ def __post_init__(self):
         elif FSDPOption.FULL_SHARD in self.fsdp and FSDPOption.SHARD_GRAD_OP in self.fsdp:
             raise ValueError("`--fsdp full_shard` is not compatible with `--fsdp shard_grad_op`.")
 
+        if FSDPOption.FULL_SHARD in self.fsdp and self.gradient_checkpointing:
```

Comment:
Is this only the case when we have FSDPOption.FULL_SHARD ?

Reply:
@SunMarc ah good point, hybrid shard also needs it. Update it now